service.go

```diff
@@ -50,7 +51,8 @@ func (*CredentialsFromStruct) IsExpired() bool {
 
 // SQSService is the service which manages a service queue on the AWS.
 type SQSService struct {
-	serviceState
+	sync.RWMutex
```

Comment:
Use an unexported attribute otherwise we will be able to lock and unlock at will from other packages.

```suggestion
	m sync.RWMutex
```